Tests/Runtime/Client/BugSplatManagerTest.cs

```diff
@@ -0,0 +1,98 @@
+using BugSplatUnity.Runtime.Client;
+using NUnit.Framework;
+using System;
+using System.Reflection;
+using UnityEditor;
+using UnityEngine;
+
+namespace BugSplatUnity.RuntimeTests.Client
+{
+    public class BugSplatManagerTest
+    {
+        private readonly string configureBugSplatMethodName = "ConfigureBugSplat";
+        private readonly string bugSplatOptionsVariableName = "bugSplatOptions";
+
+        private MethodInfo configureBugSplatMethod;
+        private BugSplatManager bugSplatManager;
+        private Type scriptType;
+
+        [SetUp]
+        public void Setup()
+        {
+            GameObject rootGameObject = new GameObject();
+            bugSplatManager = rootGameObject.AddComponent<BugSplatManager>();
+            scriptType = bugSplatManager.GetType();
+            configureBugSplatMethod = scriptType.GetMethod(configureBugSplatMethodName, BindingFlags.NonPublic | BindingFlags.Instance);
```

Comment:
If you haven't run into C# reflection before:
https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/reflection

Using reflection to invoke private methods outside of unit tests would absolutely be a cardinal sin, but given the nature of `BugSplatManager` being a `MonoBehaviour` with only `private` methods combined with the difficulty of solving this through injenction/refactoring (by bloating the number of classes needed) this seems like a fine solution for testing-sake.

Reply:
This was refactored to use `BugSplatManagerImpl`. Note the `BugSplatManagerTests` have been removed and `BugSplatManagerImpl` have been added. Additionally, we were able to move away from using reflection entirely, which I dig a lot. I think that the layer of abstraction was definitely worth the simplified unit tests.